File: 3225-length-of-longest-subarray-with-at-most-k-frequency/length-of-longest-subarray-with-at-most-k-frequency.py
class Solution:
    def maxSubarrayLength(self, nums: List[int], k: int) -> int:
        ans = 0
        n = len(nums)
        dic = defaultdict(int)
        l=0
        for r in range(n):
            dic[nums[r]] += 1
            while dic[nums[r]] > k:
                dic[nums[l]] -= 1
                l += 1
            ans = max(ans, r-l+1)
        return ans

File: 3225-length-of-longest-subarray-with-at-most-k-frequency/length-of-longest-subarray-with-at-most-k-frequency.cpp
class Solution {
public:
    int maxSubarrayLength(vector<int>& nums, int k) {
        int n = nums.size();
        unordered_map<int,int> mpp;
        int ans = 0;
        int l = 0;
        for(int r = 0; r < n; r++){
            mpp[nums[r]]+=1;
            while (mpp[nums[r]] > k){
                mpp[nums[l]] -= 1;
                l += 1;
            }
            ans = max(ans, r-l+1);
        }
        return ans;
    }
};